subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -367,6 +368,27 @@ void ztest_test_skip(void)
 	}
 }
 
+void ztest_test_expect_fail(void)
+{
+	failed_expectation = true;
+
+	switch (phase) {
+	case TEST_PHASE_SETUP:
+		PRINT(" at %s function\n", get_friendly_phase_name(phase));
+		break;
+	case TEST_PHASE_BEFORE:
+	case TEST_PHASE_TEST:
+		PRINT(" at %s function\n", get_friendly_phase_name(phase));
```

Comment:
Nit: This is the exact same thing as line 377, right? Could you just fall through to here?